src/chai-general-form-panel.ts

```diff
@@ -0,0 +1,568 @@
+      );
+    }
+  }
+
+  private handleBackgroundChange(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.background = input.value;
+    input.style.setProperty(
+      '--input-text-color',
+      this.getTextColor(this.background)
+    );
+    if (this.targetElement) {
+      this.targetElement.style.setProperty(
+        '--chai-form-background',
+        this.background
+      );
+    }
+  }
+
+  private handleFontSizeChange(e: Event) {
+    const fontSize = (e.target as HTMLInputElement).value;
+    this.fontSize = fontSize + 'px';
+    if (this.targetElement) {
+      this.targetElement.style.setProperty(
+        '--chai-form-font-size',
+        this.fontSize
+      );
+    }
+  }
+
+  private handleFontWeightChange(e: Event) {
+    const fontWeight = (e.target as HTMLInputElement).value;
+    this.fontWeight = parseInt(fontWeight);
+    if (this.targetElement) {
+      this.targetElement.style.setProperty(
+        '--chai-form-font-weight',
+        this.fontWeight.toString()
+      );
+    }
+  }
+
+  private handleCornerRadiusChange(e: Event) {
+    const cornerRadius = (e.target as HTMLInputElement).value;
+    this.cornerRadius = cornerRadius + 'px';
+    if (this.targetElement) {
+      this.targetElement.style.setProperty(
+        '--chai-form-corner-radius',
+        this.cornerRadius
+      );
+    }
+  }
+
+  private handleSpacingChange(e: Event) {
+    const spacing = (e.target as HTMLInputElement).value;
+    this.spacing = spacing + 'px';
+    if (this.targetElement) {
+      this.targetElement.style.setProperty('--chai-form-spacing', this.spacing);
+    }
+  }
+  private handleBorderWidthChange(e: Event) {
+    const width = (e.target as HTMLInputElement).value;
+
+    const value = this.border;
+    const splitedValue = value.split(' ');
+    this.border = `${width}px ${splitedValue[1]} ${splitedValue[2]}`;
+    if (this.targetElement) {
+      this.targetElement.style.setProperty('--chai-form-border', this.border);
+    }
+  }
+
+  private handleBorderColorChange(e: Event) {
+    const input = e.target as HTMLInputElement;
+    const color = input.value;
+    const value = this.border.split(' ');
+    this.border = `${value[0]} ${value[1]} ${color}`;
+
+    input.style.setProperty(
+      '--input-text-color',
+      this.getTextColor(this.border.split(' ')[2])
+    );
+    if (this.targetElement) {
+      this.targetElement.style.setProperty('--chai-form-border', this.border);
+    }
+  }
+
+  private handleBorderStyleChange(e: Event) {
+    const style = (e.target as HTMLSelectElement).value;
+    const value = this.border.split(' ');
+    this.border = `${value[0]} ${style} ${value[2]}`;
+    if (this.targetElement) {
+      this.targetElement.style.setProperty('--chai-form-border', this.border);
+    }
+  }
+
+  private handleClassNameChange(value: string) {
+    if (this.targetElement) {
+      this.targetElement.className = value;
+    }
+  }
+  private handleHeaderTextChange(value: string) {
```

Comment:
The header text is one example of a value that doesn't get reset properly when clicking "Reset Values"; if I clear it out, the displayed form actually uses the blank value (even though the blank value isn't shown in the HTML snippet), so the default header value isn't shown.